pkg/validate/serialize.go

```diff
@@ -17,28 +33,184 @@ func Verify(yamlFileName string) error {
 	}
 
 	// Value returned is a marshaled go type (CSV Struct).
-	_, err = unmarshal(rawYaml)
+	csv, err := unmarshal(rawYaml)
 	if err != nil {
 		return fmt.Errorf("Error unmarshalling YAML to OLM's csv type for %s file:  #%s ", yamlFileName, err)
 	}
 
-	fmt.Printf("%s is verified", yamlFileName)
-	return nil
+	// Contains error logs for all missing optional and mandatory fields.
+	errorLog := csvInspect(csv)
+	checkManifestResult(errorLog.warnings)
+	checkManifestResult(errorLog.errors)
+
+	// There is no mandatory field thats missing if errorLog.errors is nil.
+	if errorLog.errors != nil {
+		return fmt.Errorf("Populate all the mandatory fields missing from %s file.", yamlFileName)
+	} else {
+		fmt.Printf("%s is verified.", yamlFileName)
+		return nil
+	}
+}
+
+// Iterates over the list of warnings and errors.
+func checkManifestResult(err []missingTypeError) {
```

Comment:
not sure this is the right name for what this is doing

Reply:
How about `getErrorsFromManifestResult`?

Reply:
yeah!

Reply:
Done.